File: front-end/src/app/models/stop.ts
import {Deserializable} from "./deserializable";
import {Ride} from "./ride";
import {Order} from "./order";

export class Stop implements Deserializable {
  private _id: number;
  private _stopNumber: number;
  private _postcode: string;
  private _houseNumber: string;
  private _hasBeenVisited: boolean;
  private _ride: Ride;
  private _order: Order;
  deserialize(input: any): this {
    return Object.assign(this, input);
  }

  get id(): number {
    return this._id;
  }

  get stopNumber(): number {
    return this._stopNumber;
  }

  get postcode(): string {
    return this._postcode;
  }

  get houseNumber(): string {
    return this._houseNumber;
  }

  get hasBeenVisited(): boolean {
    return this._hasBeenVisited;
  }

  set id(value: number) {
    this._id = value;
  }

  set stopNumber(value: number) {
    this._stopNumber = value;
  }

  set postcode(value: string) {
    this._postcode = value;
  }

  set houseNumber(value: string) {
    this._houseNumber = value;
  }

  set hasBeenVisited(value: boolean) {
    this._hasBeenVisited = value;
  }

  get ride(): Ride {
    return this._ride;
  }

  set ride(value: Ride) {
    this._ride = value;
  }

  get order(): Order {
    return this._order;
  }

  set order(value: Order) {
    this._order = value;
  }
}

File: springtestproject/src/main/java/nl/vitas/springtestproject/entities/stop/Stop.java
package nl.vitas.springtestproject.entities.stop;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import nl.vitas.springtestproject.entities.order.Order;
import nl.vitas.springtestproject.entities.ride.Ride;


@Entity
@Table(name = "stops")
public class Stop {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private Integer stopNumber;

    private String postcode;

    private String houseNumber;

    private Boolean hasBeenVisited = false;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY ,cascade = CascadeType.DETACH)
    private Ride ride;


    @ManyToOne(fetch = FetchType.LAZY ,cascade = CascadeType.DETACH)
    private Order order;

    public Stop(Integer stopNumber, String postcode, String houseNumber) {
        this.stopNumber = stopNumber;
        this.postcode = postcode;
        this.houseNumber = houseNumber;
    }

    public Stop() {

    }

    public Long getId() {
        return id;
    }


    public Integer getStopNumber() {
        return stopNumber;
    }

    public String getPostcode() {
        return postcode;
    }

    public String getHouseNumber() {
        return houseNumber;
    }

    public void setStopNumber(Integer stopNumber) {
        this.stopNumber = stopNumber;
    }

    public void setPostcode(String postcode) {
        this.postcode = postcode;
    }

    public void setHouseNumber(String houseNumber) {
        this.houseNumber = houseNumber;
    }


    public Ride getRide() {
        return new Ride(ride.getRideNumber(), ride.getTimeStamp(), ride.getDescription(), ride.getStops());
    }

    public void setRide(Ride rides) {
        this.ride = rides;
    }

    public Order getOrder() {
        return new Order(order.getOrderNumber(), order.getDescription());
    }

    public void setOrder(Order order) {
        this.order = order;
    }

    public Boolean getHasBeenVisited() {
        return hasBeenVisited;
    }

    public void setHasBeenVisited() {
        this.hasBeenVisited = true;
    }

}
